shell/plugins/logging.js

```diff
@@ -0,0 +1,37 @@
+function setDebug(isDebug = false) {
+  const timestamp = function() {};
+
+  timestamp.toString = function() {
+    const ts = new Date();
+    const pad = '000';
+    const ms = ts.getMilliseconds().toString();
+    const timestamp = `${ ts.toLocaleTimeString('de-DE') }.${ pad.substring(0, pad.length - ms.length) }${ ms }: `;
+
+    return timestamp;
+  };
+
+  if (isDebug) {
+    window.debug = {
+      log:   window.console.log.bind(window.console, '%s', timestamp),
+      error: window.console.error.bind(window.console, '%s', timestamp),
+      info:  window.console.info.bind(window.console, '%s', timestamp),
+      warn:  window.console.warn.bind(window.console, '%s', timestamp)
+    };
+  } else {
+    const noOp = function() {};
+
+    window.debug = {
+      log:   noOp,
+      error: noOp,
```

Comment:
Should we only do this for console.log ? If a user has problems, we can't access them to take a look at the console logs for clues in understanding what might be happening.

Reply:
There's also problems with error monitoring tools, if we would ever want to use any. This would cause a lot of headaches as it will not collect the error at all.
All the logs will also not be highlighted as well, but that's secondary.